src/sql/parts/objectExplorer/viewlet/serverTreeDataSource.ts

```diff
@@ -76,8 +75,8 @@ export class ServerTreeDataSource implements IDataSource {
 					this._objectExplorerService.resolveTreeNodeChildren(node.getSession(), node).then(() => {
 						resolve(node.children);
 					}, expandError => {
-						element.setExpandedState(TreeItemCollapsibleState.Collapsed);
```

Comment:
Does any of this need to also be done for the `element instanceof ConnectionProfile` section above? Probably not as September work in any case but I've seen weird behavior when expanding connections times out too

Reply:
I have a comment in treeModel.ts that suggest that this is something that we need to investigate. I will add a comment to this file as well.

Reply:
Comment Added.